src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -133,6 +134,7 @@ protected Lower(Context context) {
         Preview preview = Preview.instance(context);
         useMatchException = Feature.PATTERN_SWITCH.allowedInSource(source) &&
                             (preview.isEnabled() || !preview.isPreview(Feature.PATTERN_SWITCH));
+        useDynamicConstants = target.hasDynamicConstants();
```

Comment:
I think you should allow controlling this with an option like https://github.com/openjdk/jdk/blob/5342f58dc23e9d07b0651fc28c2679db4aad27ad/src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/StringConcat.java#L83 and add a javac flag like https://github.com/openjdk/jdk/blob/ad7a8e86e0334390f87ae44cf749d2b47f1409a1/make/modules/java.base/Java.gmk#L30
so that java.base does not use dynamic constants for

Reply:
Done, thanks.

It makes sense that string concatenation would have run into similar issues. I was initially hoping it might be possible to avoid having multiple 'strategies' for enums, but it looks like it may not be.

Reply:
We must keep the old strategy for older source compatibilities before introduction of condy in release 11.